include/grpcpp/impl/codegen/server_callback.h

```diff
@@ -428,16 +391,85 @@ class ServerWriteReactor : public internal::ServerReactor {
   ServerCallbackWriter<Response>* writer_;
 };
 
+/// \a ServerUnaryReactor is a reactor-based interface for a unary RPC.
+template <class Request, class Response>
+class ServerUnaryReactor : public internal::ServerReactor {
+ public:
+  ~ServerUnaryReactor() = default;
+
+  /// The following operation initiations are exactly like ServerBidiReactor.
+  void StartSendInitialMetadata() { call_->SendInitialMetadata(); }
+  void Finish(Status s) { call_->Finish(std::move(s)); }
+
+  /// Similar to ServerBidiReactor::OnStarted, except that this also provides
+  /// the request object sent by the client and the response object that the
+  /// stream fills in before calling Finish. (It must be filled in if status
+  /// is OK, but it may be filled in otherwise.)
+  ///
+  /// \param[in] req The request object sent by the client
+  /// \param[in] resp The response object to be used by this RPC
+  virtual void OnStarted(const Request* req, Response* resp) {}
+
+  /// The following notifications are exactly like ServerBidiReactor.
+  virtual void OnSendInitialMetadataDone(bool ok) {}
+  void OnDone() override {}
+  void OnCancel() override {}
+
+  // NOTE: This is an API for advanced users who need custom allocators.
+  // Get and maybe mutate the allocator state associated with the current RPC.
+  RpcAllocatorState* GetRpcAllocatorState() {
+    return call_->GetRpcAllocatorState();
+  }
+
+ private:
+  friend class ServerCallbackUnary;
+  void BindCall(ServerCallbackUnary* call) { call_ = call; }
+
+  ServerCallbackUnary* call_;
+};
+
+template <typename Request, typename Response, typename F>
+ServerUnaryReactor<Request, Response>* ServeRpc(ServerContext* context,
```

Comment:
I prefer MakeReactor like Sanjay suggested

Reply:
Or maybe MakeServerUnaryReactor

Reply:
Made it MakeReactor.